sqlwhat/check_funcs.py

```diff
@@ -37,24 +64,17 @@ def check_clause(name, missing_msg="missing clause", state=None):
 
     return state.to_child(student_ast = stu_attr, solution_ast = sol_attr)
 
-def check_correct(index, state=None):
+def check_correct(state, index):
     pass
 
-def check_result(msg="Incorrect result.", state=None):
-    stu_res = state.student_result
-    sol_res = state.solution_result
-    if stu_res != sol_res:
-        state.reporter.do_test(Test(msg))
-
-    return state
-
-def has_equal_ast(msg="Incorrect AST", state=None):
-    if repr(state.student_ast) != repr(state.solution_ast):
+def has_equal_ast(state, msg="Incorrect AST", sql=None, start="sql_script"):
```

Comment:
I love the option for setting the `sql`! At some point, it would be nice to make the `override_sql` option part of the chain of SCT commands, I think. That way, if you have other SCT functions that could benefit from the overriding option without having to include the argument and logic everywhere.